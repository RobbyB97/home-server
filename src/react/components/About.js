import React from 'react';

import Divider from './Divider'


export class About extends React.Component {
    render() {
        return (
            <section id="About" className="about">

                <div className="about__content one">
                    <div className="about__portrait">
                        <img src="/dist/images/aboutImage.png" />
                        <div className="about__text one">
                            <p>
                                <span className="one">Hi, my name's Robby. <span className="two">I'm a front-end developer and digital artist</span>.</span>
                            </p>
                        </div>
                    </div>
                </div>

                <Divider section="about two" />

                <div className="about__content two">
                    <div className="about__text two">
                        <p>
                            I want to help you succeed by building a beautiful, fast and functional website for your business.
                        </p>
                    </div>
                </div>
            </section>
        );
    }
}


export default About;