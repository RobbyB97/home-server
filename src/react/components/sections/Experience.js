import React from 'react';


export class Experience extends React.Component {
    render() {
        return (
            <section id="Experience" className="experience">
                
            </section>
        );
    }
}


export default Experience;