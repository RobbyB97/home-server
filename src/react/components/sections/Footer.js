import React from 'react';

import Divider from '../Divider'


export class Footer extends React.Component {
    render() {
        return (
            <footer id="Footer" className="footer">
                <Divider section="footer__divider"/>
                
                <div className="footer__content">
                    <div className="footer__links">
                        
                    </div>

                    <p className="footer__copyright">
                        Â© Robert Bergers 2017-2020
                    </p>
                </div>
            </footer>
        );
    }
}


export default Footer;