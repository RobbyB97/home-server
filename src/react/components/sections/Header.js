import React from 'react';

import DesktopNavigation from '../ui/DesktopNavigation';
import MobileNavigation from '../ui/MobileNavigation';
import HomeButton from '../ui/HomeButton';



export class Header extends React.Component {
    render() {
        return (
            <header id="Header" className="header">
                <HomeButton internal />
                <DesktopNavigation />
                <MobileNavigation />
            </header>
        );
    }
}


export default Header;