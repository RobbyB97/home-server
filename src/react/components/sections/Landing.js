import React from 'react';
import Particles from 'react-particles-js';


export class Landing extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            emailModal: false
        };
    }

    componentDidMount() {
        setTimeout(() => {
            const thing = document.querySelector('p.left');
            
        }, 1000)
    }

    /* Controls email modal */
    toggleEmail() {
        this.setState({
            emailModal: !this.state.emailModal
        });
    }

    render() {
        return (
            <section id="Landing" className="landing">
                {/* Background color blocks */}
                <div className="landing__background">
                    <div className="one">
                        <div className="texture">
                            <div></div>
                        </div>
                    </div>

                    <div className="two">
                        <div className="texture">
                            <div></div>
                        </div>
                    </div>

                    <div className="three"></div>
                    
                    <div className="four">
                        <Particles 
                            width="100vw"
                            height="100vh"
                            className="landing__particles"
                            params={{
                                particles: {
                                    number: {
                                        value: 110,
                                        density: {
                                            enable: true,
                                            value_area: 700
                                        }
                                    },
                                    line_linked: {
                                        enable: true,
                                        distance: 200,
                                        color: '#f8f8f8',
                                        opacity: 0.5,
                                        width: 1
                                    },
                                    color: {
                                        value: '#f8f8f8'
                                    }
                                }
                            }}
                        />
                    </div>
                    
                    <div className="five"></div>
                </div>
                    
                {/* Tagline */}
                <header className="landing__tagline">
                    <h2 className="left">
                        Robby
                    </h2>

                    <h2 className="right">
                        Bergers
                    </h2>

                    <p className="left">
                        Designer
                    </p>

                    <p className="right">
                        Developer
                    </p>
                </header>

                {/* Link icons */}
                <nav className="landing__contact">
                    <a className="landing__github" href="https://www.github.com/robbyb97">
                        <div></div>
                    </a>
                    
                    <a className="landing__linkedin" href="https://www.linkedin.com/in/robbyb97/">
                        <div></div>
                    </a>

                    <a className="landing__upwork" href="https://www.upwork.com/o/profiles/users/~01f696e65011a54866/">
                        <div></div>
                    </a>

                    <a className="landing__twitter" href="https://twitter.com/RobbyBergers">
                        <div></div>
                    </a>
                </nav>

                {/* Menu buttons */}
                <nav className="landing__menu">
                    <a className="landing__view" href="#About">
                        See my work
                    </a>

                    <a className="landing__resume" target="_blank" href="https://bergers.dev/resume.pdf">
                        View my resume
                    </a>
                    
                    <a className="landing__email"
                        onClick={() => {
                            this.toggleEmail();
                        }}
                    >
                        Get in touch
                    </a>
                </nav>

                {/* Email modal */}
                <div 
                    className="landing__emailModal"
                    data-active={this.state.emailModal}
                >
                    <p>Email me @ <a href="mailto:bergersr97@gmail.com">bergersr97@gmail.com</a></p>
                </div>
            </section>
        );
    }
}


export default Landing;