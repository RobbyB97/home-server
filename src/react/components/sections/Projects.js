import React from 'react';


export class Projects extends React.Component {
    render() {
        return (
            <section id="Projects" className="projects">
                
            </section>
        );
    }
}


export default Projects;