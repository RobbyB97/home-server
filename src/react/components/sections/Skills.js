import React from 'react';

import SkillSection from '../SkillSection';


export class Skills extends React.Component {
    render() {
        const sections = [
            {
                title: "Front end",
                skills: [
                    "React", "SASS", "HTML5", "CSS3", "JS (ES6)"
                ]
            }, {
                title: "Workflow",
                skills: [
                    "Git", "NPM", "Gulp", "Webpack", "Jest", 
                    "Firebase", "Heroku"
                ]
            }, {
                title: "Back end",
                skills: [
                    "Node.js", "BASH", "Linux/Ubuntu server",
                    "MongoDB", "Oracle SQL"
                ]
            }
        ];

        return (
            <section id="Skills" className="skills">
                <header className="skills__header">
                    <h2>My skillset</h2>
                </header>

                {sections.map((section) => (
                    <SkillSection 
                        {...section}
                        key={section.title}
                    />
                ))}
            </section>
        );
    }
}


export default Skills;