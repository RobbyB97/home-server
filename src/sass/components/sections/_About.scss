#About {}


.about__content {
    display: inline-block;
    font-family: $montserrat;
    text-align: center;
    width: 100%;
    padding-bottom: 20px;

    &.one {
        background-color: lighten($black, 15%);
    }
}


.about__bioTitle {
    font-size: 1.4em;
    font-weight: bold;
}


.about__portrait {
    margin: 0 auto;
    img {
        width: 300px;
        height: auto;
        border-radius: 50%;

        @media #{$bp-tablet} {
            width: 400px;
        }

        @media #{$bp-desktop} {
            width: 500px;
        }
    }
}


.about__text {
    width: 100%;
    font-size: 1.1em;
    
    p {
        width: 280px;
        margin: 0 auto;
    }
}


.about__text.one {
    position: absolute;

    p {
        position: relative;
        bottom: 60px;

        span.one {
            padding: 1px;
            background-color: $black;
        }

        span.two {
            padding: 1px;
            font-size: 1.2em;
            background-color: $red;
        }
    }

    @media #{$bp-ultrawide} {
        p {
            padding-top: 10px;
        }
    }
}


.about__divider.one {
    height: 0;

    .divider__shape {
        &.one {
            border-top: 30px solid lighten($black, 15%);
        }

        &.two {
            display: none;
        }
    }

    @media #{$bp-tablet} {
        .divider__shape {
            &.one {
                border-top: 45px solid lighten($black, 15%);
            }
        }
    }

    @media #{$bp-desktop} {
        .divider__shape {
            &.one {
                border-top: 60px solid lighten($black, 15%);
            }
        }
    }

    @media #{$bp-ultrawide} {
        .divider__shape {
            &.one {
                border-top: 90px solid lighten($black, 15%);
            }
        }
    }
}


.about__content.two {
    background: lighten($black, 10%);
    padding-bottom: 0;
}


.about__text.two {
    padding-top: 5vh;

    @media #{$bp-ultrawide} {
        p {
            margin-top: 10px;
        }
    }
}


.about__divider.two {
    .divider__shape {
        &.one {
            border-top: 30px solid lighten($black, 10%);
        }

        &.two {
            display: none;
        }
    }

    @media #{$bp-tablet} {
        .divider__shape {
            &.one {
                border-top: 45px solid lighten($black, 10%);
            }
        }
    }

    @media #{$bp-desktop} {
        .divider__shape {
            &.one {
                border-top: 60px solid lighten($black, 10%);
            }
        }
    }

    @media #{$bp-ultrawide} {
        .divider__shape {
            &.one {
                border-top: 90px solid lighten($black, 10%);
            }
        }
    }
}