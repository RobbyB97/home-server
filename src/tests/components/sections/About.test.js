import React from 'react'
import {shallow} from 'enzyme'

import {About} from '../../../react/components/sections/About'


test('Render About', () => {
    const wrapper = shallow(<About />)
    expect(wrapper).toMatchSnapshot()
})