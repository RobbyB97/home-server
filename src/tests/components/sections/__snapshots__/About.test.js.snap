// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render About 1`] = `
<section
  className="about"
  id="About"
>
  <div
    className="about__content one"
  >
    <figure
      className="about__portrait"
    >
      <img
        src="/dist/images/aboutImage.png"
      />
      <div
        className="about__text one"
      >
        <p>
          <span
            className="one"
          >
            Hi, my name's Robby. 
            <span
              className="two"
            >
              I'm a front-end developer and digital artist
            </span>
            .
          </span>
        </p>
      </div>
    </figure>
  </div>
  <Divider
    section="about__divider one"
  />
  <div
    className="about__content two"
  >
    <div
      className="about__text two"
    >
      <p>
        I want to help you succeed by building a beautiful, fast and functional website for your business.
      </p>
    </div>
  </div>
  <Divider
    section="about__divider two"
  />
</section>
`;
